### PR TITLE
Fixes for distributed and state tests

### PR DESCRIPTION
**Pull request by Shillaker**
Changes:

- Add state distributed test using `demo/pi` function (distributed Pi calculation).
- Add tests running OpenMP and MPI Pi calculations.
- Re-enable and fix all OpenMP tests.

This requires these two PRs to be merged in before closing: 

- https://github.com/faasm/faabric/pull/241
- https://github.com/faasm/cpp/pull/84

### CODE REVIEW

File: src/wavm/mpi.cpp [hunk 1]
```diff
@@ -18,10 +18,10 @@ using namespace faabric::scheduler;
 using namespace WAVM;
 
 #define MPI_FUNC(str)                                                          \
-    SPDLOG_DEBUG("MPI-{} {}", executingContext.getRank(), str);
+    SPDLOG_TRACE("MPI-{} {}", executingContext.getRank(), str);
```
Comment (Shillaker) [inline src/wavm/mpi.cpp:21]: These were printing out so much logging information in the dist tests that it was difficult to debug.

File: tests/test/faaslet/test_mpi.cpp [hunk 2]
```diff
@@ -132,4 +134,11 @@ TEST_CASE_METHOD(MPIFuncTestFixture, "Test MPI async", "[mpi]")
 {
     checkMpiFunc("mpi_isendrecv");
 }
+
+TEST_CASE_METHOD(MPIFuncTestFixture, "Test MPI Pi", "[mpi]")
+{
+    faabric::Message result = checkMpiFunc("mpi_pi");
+    std::string output = result.outputdata();
+    REQUIRE(faabric::util::startsWith(output, "Pi estimate: 3.1"));
+}
```
Comment (Shillaker) [inline tests/test/faaslet/test_mpi.cpp:143]: Noticed that this wasn't included in tests so I added it for good measure.

File: docs/source/development.md [hunk 3]
```diff
@@ -219,55 +219,47 @@ interfering with the tests:
 docker-compose down
 ```
 
-Start the distributed tests server:
+Make sure your local setup is built, along with the distributed tests:
```
[the remark this answers is not in the document]
Comment (csegarragonz) [reply docs/source/development.md:222]: Agree! :rofl:

File: tests/dist/threads/test_openmp.cpp [hunk 4]
```diff
@@ -3,12 +3,14 @@
 #include "fixtures.h"
 
 #include <faabric/scheduler/Scheduler.h>
+#include <faabric/util/string_tools.h>
+
+#define PI_FUNCTION "pi_faasm"
```
Comment (csegarragonz) [inline tests/dist/threads/test_openmp.cpp:8]: Very minor, but it feels a bit odd that this is the only function with its name `#define`-ed and also that it follows a different format than MPI. I would have expected it to be called `omp_pi`.

**Review by csegarragonz — APPROVED** [review]
Nice to see the troublesome `omp_static_for` is back!

PS: friendly reminder to update the submodules before merging this in.

**Comment by Shillaker** [comment]
> Nice to see the troublesome `omp_static_for` is back!

Haha yes, wonder how long it will survive for this time :smile:

**Pull request merged**